named_entity_recognition/named_entity_recognition_question_answering.ipynb

```diff
@@ -0,0 +1,1341 @@
+    "            data_links_dict[subset_name].append(obj['Key'])\n",
+    "        print(f\"\\t{len(data_links_dict[subset_name])} files in {subset_name}\")\n",
+    "        start_key = objects[-1]['Key']\n",
+    "        if len(objects) < 1000:\n",
+    "            start_key = ''\n",
+    "            break"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c191650a",
+   "metadata": {},
+   "source": [
+    "### 2.2. Download files\n",
+    "<a id='download_files'></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "892942fe-9b07-4e01-b0d1-b8c4b60920fb",
+   "metadata": {},
+   "source": [
+    "Now we will use these links to download all the files from S3 bucket."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "216dfcbf",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for subset_name in subset_names:\n",
+    "    print(f\"Loading {subset_name} docs\")\n",
+    "    save_dir = f'./{subset_name}_sa_docs'\n",
+    "    if not os.path.exists(save_dir):\n",
+    "        os.mkdir(save_dir)\n",
+    "    for file_key in tqdm(data_links_dict[subset_name]):\n",
+    "        if not '.txt' in file_key:\n",
+    "            continue\n",
+    "        filename = os.path.basename(file_key)\n",
+    "        s3_client.download_file(Bucket=bucket_name, \n",
+    "                                Key=file_key,\n",
+    "                                Filename=os.path.join(save_dir, filename))\n",
+    "        "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "6e36049c",
+   "metadata": {},
+   "source": [
+    "### 2.3 Download labels from SuperAnnotate\n",
+    "<a id='download_labels_from_sa'></a>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "4c569da6-7eb9-4a62-afef-8764aa2fce1a",
+   "metadata": {},
+   "source": [
+    "Now we can download labels from SuperAnnotate for the train texts that were annotated manually. The annotations will be downloaded in [SuperAnnotate format](https://doc.superannotate.com/docs/sdk-export-annotations)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "bff0e565",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "token = \"PUT_YOUR_TOKEN_HERE\"\n",
+    "\n",
+    "sa_client = SAClient(token = token)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "e13e258f",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sa_response = sa_client.get_annotations(project=SA_PROJECT_NAME,\n",
+    "                                        items=[os.path.basename(x) for x \\\n",
+    "                                               in data_links_dict['train']])\n",
+    "\n",
+    "annotations = [i['instances'] for i in sa_response]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "58d74089",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "prompt_mapper = {'CASE_NUMBER': 'case number',\n",
```

Comment:
add extra \t under, so it will look more structured

Reply:
+